templates/infra.json

```diff
@@ -8,340 +8,426 @@
-        "functionAppName-engine": "[concat(parameters('functName'), '-engine')]",
-        "hostingPlanName": "[concat(parameters('appName'), '-asp')]",
-        "applicationInsightsName": "[concat(parameters('appName'), '-appinsights')]",
-        "appConfigName": "[concat(parameters('appName'), '-appconfig')]",
-        "scalerFuncWorkerRuntime": "[parameters('scalerRuntime')]",
-        "engineFuncWorkerRuntime": "[parameters('engineRuntime')]",
-        "stgQueueName": "[concat(parameters('stgacctName'), '/default/', parameters('queueName'))]"
+        "contributorRole": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')]"
     },
     "resources": [
         {
-            "type": "Microsoft.Storage/storageAccounts",
-            "apiVersion": "2019-06-01",
-            "name": "[parameters('stgacctName')]",
-            "location": "[parameters('location')]",
-            "sku": {
-                "name": "[parameters('storageAccountType')]"
-            },
-            "kind": "Storage",
-            "resources": [
-            ]
+            "type": "Microsoft.Resources/resourceGroups",
+            "apiVersion": "2020-06-01",
+            "name": "[concat(parameters('appName'), '-rg')]",
+            "location": "[deployment().location]",
+            "properties": {}
         },
         {
-            "type": "Microsoft.Storage/storageAccounts/queueServices",
-            "apiVersion": "2020-08-01-preview",
-            "name": "[concat(parameters('stgacctName'), '/default')]",
+            "type": "Microsoft.Authorization/roleAssignments",
+            "apiVersion": "2020-04-01-preview",
+            "name": "[parameters('engineContributorGUID')]",
             "dependsOn": [
-                "[resourceId('Microsoft.Storage/storageAccounts', parameters('stgacctName'))]"
+                "bellhopDeploy"
             ],
             "properties": {
-                "cors": {
-                    "corsRules": [
-                    ]
-                }
+                "scope": "[subscription().id]",
+                "roleDefinitionId": "[variables('contributorRole')]",
+                "principalId": "[reference('bellhopDeploy').outputs.enginePrincipalId.value]"
             }
         },
         {
-            "type": "Microsoft.Storage/storageAccounts/queueServices/queues",
-            "apiVersion": "2019-06-01",
-            "name": "[variables('stgQueueName')]",
+            "type": "Microsoft.Authorization/roleAssignments",
+            "apiVersion": "2020-04-01-preview",
+            "name": "[parameters('scalerContributorGUID')]",
             "dependsOn": [
-                "[resourceId('Microsoft.Storage/storageAccounts', parameters('stgacctName'))]",
-                "[resourceId('Microsoft.Storage/storageAccounts/queueServices', parameters('stgacctName'), 'default')]"
+                "bellhopDeploy"
```

Comment:
Same as before: Not important, but since this depends on bellhopDeploy, I would encourage you to put this below that object in the ARM template. Improves readability.

Reply:
Makes sense, can do (in a future update).